### DISCUSSION TITLE
How to modify rememberLineCartesianLayer when series change

### DISCUSSION THREAD

**Discussion by ryancstack** [patrykandpatrick/vico#935] [Questions]
### Question

I have a graph with multiple series where I can toggle some on and off. However, whenever I create a new set of lines, they don't update on the graph in that the series maps to a stale cached color:

```
           rememberLineCartesianLayer(
              LineCartesianLayer.LineProvider.series(
                historicPrices
                  .filter { it.condition in enabledConditions }
                  .mapIndexed { index, dataPoint ->
                    val color = colorMapping[dataPoint.condition] ?: Color.DarkGray
                    LineCartesianLayer.rememberLine(
                      fill = remember { LineCartesianLayer.LineFill.single(fill(color)) },
                      areaFill = null,
                    )
                }
              )
            )
```   

enabledConditions is a set that tracks the toggles. 

On another topic, is there a way to add click listeners to the built in legend support?

### Vico version(s)

2.0.0-beta.2

### UI framework(s)

Jetpack Compose

**Comment by ryancstack** [comment]
I managed to get a working version following what was suggested here https://github.com/patrykandpatrick/vico/discussions/924

```
ExtraStoreLineProvider.series { seriesIndex, extraStore ->
  // .. get rememberLine based on seriesIndex and some other factors derived from extraStore
}
```

```
data class ExtraStoreLineProvider(private val generator: (seriesIndex: Int, extraStore: ExtraStore) -> Line) : LineProvider {
  override fun getLine(
    seriesIndex: Int,
    extraStore: ExtraStore
  ): Line {
    return generator(seriesIndex, extraStore)
  }

  companion object {
    fun series(generator: (seriesIndex: Int, extraStore: ExtraStore) -> Line): LineProvider {
      return ExtraStoreLineProvider(generator)
    }
  }
}
```

**Reply by Gowsky** [reply]
Hello, @ryancstack.

> On another topic, is there a way to add click listeners to the built in legend support?

This isn't supported out of the box at the moment. Could you share some more details on the use case? Would you like to highlight a given line when the corresponding legend item is tapped, for example?

**Reply by ryancstack** [reply]
yes exactly. filter out lines or select lines for display

**Reply by Gowsky** [reply]
Please feel free to open a feature request in the Ideas category.
